core/src/org/pentaho/di/core/extension/ExtensionPointMap.java

```diff
@@ -90,7 +90,10 @@ public static ExtensionPointMap getInstance() {
 
   /**
    * Add the extension point plugin to the map
-   * 
+   *
+   * Warning: Do not call this method directly through ExtensionPointMap, it may lead to concurrency problems. Call
+   * PluginRegistry.getInstance().registerPlugin() instead.
+   *
    * @param extensionPointPlugin
    */
   public void addExtensionPoint( PluginInterface extensionPointPlugin ) {
```

Comment:
Ideally, yes.

However, this is public-facing code, and if we were to do such a visibility change on a minor release, we might:

- cause compiler errors on any customer/community custom code out there that may be accessing it ( i.e. custom Kettle plugins )
- cause the same issue to occur on previous versions, should this fix ever be included in a SP case

Safest approach seems to be to mark this as deprecated and leave a comment with a recommended alternative.

If you still feel otherwise, please let us know and we'll do the change.